file_store/src/lora_beacon_report.rs

```diff
@@ -33,50 +33,22 @@ impl MsgDecode for LoraBeaconIngestReport {
 impl TryFrom<LoraBeaconReportReqV1> for LoraBeaconIngestReport {
     type Error = Error;
     fn try_from(v: LoraBeaconReportReqV1) -> Result<Self> {
-        //TODO: fix this to return error is not valid value
-        // let data_rate: DataRate = DataRate::from_i32(v.datarate).unwrap();
-        let data_rate: DataRate =
-            DataRate::from_i32(v.datarate).ok_or_else(|| Error::custom("unsupported datarate"))?;
         Ok(Self {
             received_timestamp: Utc::now(),
-            report: LoraBeaconReport {
-                pub_key: PublicKey::try_from(v.pub_key)?,
-                local_entropy: v.local_entropy,
-                remote_entropy: v.remote_entropy,
-                data: v.data,
-                frequency: v.frequency,
-                channel: v.channel,
-                datarate: data_rate,
-                tx_power: v.tx_power,
-                timestamp: datetime_from_epoch(v.timestamp),
-                signature: v.signature,
-            },
+            report: v.try_into()?,
```

Comment:
nice, i had been planning on coming back to these original reports and trying something similar to this